src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -1155,12 +1157,7 @@ public function class_list() {
 				return;
 			}
 
-			/*
-			 * CSS class names are case-insensitive in the ASCII range.
-			 *
-			 * @see https://www.w3.org/TR/CSS2/syndata.html#x1
-			 */
-			$name = strtolower( substr( $class, $at, $length ) );
+			$name = str_replace( "\x00", "\u{FFFD}", substr( $class, $at, $length ) );
```

Comment:
**This needs to be extracted to its own PR for consideration.**

The browser will replace null bytes with `�` in attribute values. It's impossible to have a null byte in an attribute value that comes from HTML. It's possible to set a null byte in an attribute value via the DOM.

We may consider that `class_list` is a "processed" version of the `class` attribute, and this null-byte replacement is appropriate here. At this time, `get_attribute` is closer to "raw" access to the attribute value as authored.

Related to https://github.com/WordPress/wordpress-develop/pull/7184.

Reply:
Moved this to https://github.com/WordPress/wordpress-develop/pull/7187.

Reply:
I don't think this needs to be in a separate PR. we can entertain it wherever. A separate PR is fine though. The focus is on handling CSS class names properly, so in that regard this is purely about matching CSS class selectors to values in HTML `class` attributes.